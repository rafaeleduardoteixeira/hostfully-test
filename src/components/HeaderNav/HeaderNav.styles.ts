import styled from 'styled-components';
import { colors, media } from '../../themes';

export const Container = styled.div`
  display: flex;
  align-items: center;
  gap: 15px;
  border: 1px solid ${colors.grayLight};
  border-radius: 30px;
  padding: 0 15px;
  cursor: pointer;
  height: 45px;

  transition: box-shadow 0.2s cubic-bezier(0.2, 0, 0, 1);

  &:hover {
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.18);
  }

  @media ${media.tablet} {
    padding: 0 25px;
    height: 60px;
  }

  @media ${media.mobile} {
    padding: 0 15px;
    height: 45px;
  }
`;

export const HeaderNavIcon = styled.img`
  width: 18px;
`;

export const HeaderNavAvatar = styled.img`
  border-radius: 50%;
  object-fit: cover;
  width: 32px;
  height: 32px;
`;
