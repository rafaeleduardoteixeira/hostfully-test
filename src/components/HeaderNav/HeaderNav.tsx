import { Container, HeaderNavAvatar, HeaderNavIcon } from './HeaderNav.styles';

export const HeaderNav = (): JSX.Element => {
  return (
    <Container>
      <HeaderNavIcon src="/icons/header-nav.svg" />
      <HeaderNavAvatar
        src={
          'https://www.rafaeleteixeira.com.br/_next/image?url=%2F_next%2Fstatic%2Fmedia%2Fprofile.a4caa42b.jpg&w=3840&q=75'
        }
      />
    </Container>
  );
};
