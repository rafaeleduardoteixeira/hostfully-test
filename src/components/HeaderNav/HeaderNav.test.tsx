import React, { act } from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { HeaderNav } from './HeaderNav';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe('HeaderNav', () => {
  test('Snapshot test', () => {
    const { asFragment } = render(<HeaderNav />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('Should render HeaderNav', () => {
    render(<HeaderNav />);
    const HeaderNavElement = screen.getByTestId('HeaderNavMenuClick');
    expect(HeaderNavElement).toBeInTheDocument();
  });
  test('Should click on HeaderNav and open DropDownMenu', () => {
    render(<HeaderNav />);
    const HeaderNavMenuClick = screen.getByTestId('HeaderNavMenuClick');
    expect(HeaderNavMenuClick).toBeInTheDocument();
    act(() => {
      HeaderNavMenuClick.click();
    });
    const DropDownMenuElement = screen.getByTestId('DropDownMenu');
    expect(DropDownMenuElement).toBeInTheDocument();
  });

  test('Should click outside HeaderNav and close DropDownMenu', () => {
    render(<HeaderNav />);
    const HeaderNavMenuClick = screen.getByTestId('HeaderNavMenuClick');
    expect(HeaderNavMenuClick).toBeInTheDocument();
    act(() => {
      HeaderNavMenuClick.click();
    });
    const DropDownMenu = screen.getByTestId('DropDownMenu');
    expect(DropDownMenu).toBeInTheDocument();
    act(() => {
      document.dispatchEvent(new MouseEvent('mousedown', { bubbles: true }));
    });
    expect(DropDownMenu).not.toBeInTheDocument();
  });
});
