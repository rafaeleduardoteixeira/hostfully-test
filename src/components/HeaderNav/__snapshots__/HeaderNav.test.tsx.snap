// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNav Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="sc-braxZu cnSkOo"
    data-testid="HeaderNavMenuClick"
  >
    <img
      class="sc-gJhJTp cboMkP"
      src="/icons/header-nav.svg"
    />
    <img
      class="sc-elDIKY RUxgM"
      src="https://www.rafaeleteixeira.com.br/_next/image?url=%2F_next%2Fstatic%2Fmedia%2Fprofile.a4caa42b.jpg&w=3840&q=75"
    />
  </div>
</DocumentFragment>
`;
